File: src/main/java/com/sfh/pokeRogueBot/service/WaitingService.java
package com.sfh.pokeRogueBot.service;

import com.sfh.pokeRogueBot.config.WaitConfig;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Slf4j
@Service
public class WaitingService {

    private final WaitConfig waitConfig;

    public WaitingService(WaitConfig waitConfig) {
        this.waitConfig = waitConfig;
    }

    public void waitBriefly() {
        int waitTime = waitConfig.getWaitBriefly();
        sleep(waitTime);
    }

    public void waitModifierCursor() {
        int waitTime = waitConfig.getWaitModifierCursor();
        sleep(waitTime);
    }

    public void waitLonger() {
        int waitTime = waitConfig.getWaitLonger();
        sleep(waitTime);
    }

    public void sleep(int waitTime) {
        try {
            Thread.sleep(waitTime);
        } catch (InterruptedException e) {
            log.error("Error while waiting, error: " + e.getMessage());
        }
    }
}

File: src/main/kotlin/com/sfh/pokeRogueBot/service/WaitingService.kt
package com.sfh.pokeRogueBot.service

import com.sfh.pokeRogueBot.config.WaitConfig
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class WaitingService(private val waitConfig: WaitConfig) {

    companion object {
        private val log = LoggerFactory.getLogger(WaitingService::class.java)
    }

    fun waitBriefly() {
        sleep(waitConfig.waitBriefly)
    }

    fun waitModifierCursor() {
        sleep(waitConfig.waitModifierCursor)
    }

    fun waitLonger() {
        sleep(waitConfig.waitLonger)
    }

    fun waitForNotActiveWaveRunner() {
        sleep(waitConfig.waitForNotActiveWaverunner)
    }

    private fun sleep(waitTime: Int) {
        try {
            Thread.sleep(waitTime.toLong())
        } catch (e: InterruptedException) {
            log.error("Error while waiting, error: " + e.message)
        }
    }
}